packages/wasm-dpp/test/unit/stateTransition/StateTransitionFactory.spec.js

```diff
@@ -1,160 +1,99 @@
-const getDataContractFixture = require('@dashevo/dpp/lib/test/fixtures/getDataContractFixture');
-
-const DataContractCreateTransition = require('@dashevo/dpp/lib/dataContract/stateTransition/DataContractCreateTransition/DataContractCreateTransition');
+const protocolVersion = require('@dashevo/dpp/lib/version/protocolVersion');
 
-const ValidationResult = require('@dashevo/dpp/lib/validation/ValidationResult');
+const getDataContractFixture = require('@dashevo/dpp/lib/test/fixtures/getDataContractFixture');
 
-const InvalidStateTransitionError = require('@dashevo/dpp/lib/stateTransition/errors/InvalidStateTransitionError');
+const createStateRepositoryMock = require('@dashevo/dpp/lib/test/mocks/createStateRepositoryMock');
 
-const SerializedObjectParsingError = require('@dashevo/dpp/lib/errors/consensus/basic/decode/SerializedObjectParsingError');
+const { default: loadWasmDpp } = require('../../../dist');
+const getBlsAdapterMock = require('../../../lib/test/mocks/getBlsAdapterMock');
 
-const createDPPMock = require('@dashevo/dpp/lib/test/mocks/createDPPMock');
-const StateTransitionFactory = require('@dashevo/dpp/lib/stateTransition/StateTransitionFactory');
-const SomeConsensusError = require('@dashevo/dpp/lib/test/mocks/SomeConsensusError');
+describe('StateTransitionFactory', function main() {
+  this.timeout(100000);
 
-describe('StateTransitionFactory', () => {
-  let validateStateTransitionBasicMock;
-  let createStateTransitionMock;
   let factory;
   let stateTransition;
   let rawStateTransition;
-  let decodeProtocolEntityMock;
-  let dppMock;
+  let stateRepositoryMock;
+  let StateTransitionFactory;
+  let DataContractCreateTransition;
+
+  before(async () => {
+    ({
+      DataContractCreateTransition, StateTransitionFactory,
+    } = await loadWasmDpp());
+  });
 
-  beforeEach(function beforeEach() {
+  beforeEach(async function beforeEach() {
     const dataContract = getDataContractFixture();
 
+    const rawDataContract = dataContract.toObject();
+
     stateTransition = new DataContractCreateTransition({
-      dataContract: dataContract.toObject(),
+      protocolVersion: protocolVersion.latestVersion,
+      dataContract: rawDataContract,
       entropy: dataContract.getEntropy(),
+      signature: Buffer.alloc(65),
+      signaturePublicKeyId: 0,
     });
     rawStateTransition = stateTransition.toObject();
 
-    decodeProtocolEntityMock = this.sinonSandbox.stub();
-
-    validateStateTransitionBasicMock = this.sinonSandbox.stub();
-    createStateTransitionMock = this.sinonSandbox.stub().returns(stateTransition);
-
-    dppMock = createDPPMock();
+    stateRepositoryMock = createStateRepositoryMock(this.sinonSandbox);
+
+    stateRepositoryMock.fetchDataContract.resolves(dataContract);
+    stateRepositoryMock.createDataContract.resolves();
```

Comment:
Do we need to resolve to empty functions everywhere? I think mock by default is kinda doing that already

Reply:
@shumkov requested to get rid of default `resolve()` a while ago. They have to be provided explicitly